mpcite/doi_builder.py

```diff
@@ -0,0 +1,487 @@
+
+        :param mp_id: mp_id to query for in the robocrys database
+        :return:
+            description in string
+        """
+        robo_result = self.robocrys_store.query_one(criteria={self.robocrys_store.key: mp_id})
+        if robo_result is None:
+            return ELinkGetResponseModel.get_default_description()
+        else:
+            robo_result = RoboCrysModel.parse_obj(robo_result)
+            description = robo_result.description
+            if description is None:
+                description = ELinkGetResponseModel.get_default_description()
+            return description[:12000]  # 12000 is the Elink Abstract character limit
+
+    def get_doi(self, mp_id) -> str:
+        osti_id = self.get_osti_id(mp_id=mp_id)
+        if osti_id == '':
+            return ''
+        else:
+            return "10.17188/" + osti_id
+
+    def get_osti_id(self, mp_id) -> str:
+        """
+        Used to determine if an update is necessary.
+
+        If '' is returned, implies update is not necessary.
+
+        Otherwise, an update is necessary
+        :param mp_id:
+        :return:
+        """
+        doi_entry = self.doi_store.query_one(criteria={self.doi_store.key: mp_id})
+        if doi_entry is None:
+            return ''
+        else:
+            return doi_entry['doi'].split('/')[-1]
+
+    def as_dict(self) -> dict:
+        return {
+            "materials_collection": self.materials_store.as_dict(),
+            "robocrys_collection": self.robocrys_store.as_dict(),
+            "dois_collection": self.doi_store.as_dict(),
+            "osti": self.osti.dict(),
+            "elsevier": self.elsevier.dict(),
+            "max_doi_requests": self.max_doi_requests,
+            "sync": self.sync,
+        }
+
+    @classmethod
+    def from_dict(cls, d: dict):
+        assert "osti" in d, "Error: OSTI config not found"
+        assert "elsevier" in d, "Error: elsevier config not found"
+        assert "materials_collection" in d, "Error: materials_collection config not found"
+        assert "dois_collection" in d, "Error: dois_collection config not found"
+        assert "robocrys_collection" in d, "Error: robocrys_collection config not found"
+        assert "max_doi_requests" in d, "Error: max_doi_requests config not found"
+        assert "sync" in d, "Error: sync config not found"
+
+        elink = ConnectionModel.parse_obj(d["osti"]["elink"])
+        explorer = ConnectionModel.parse_obj(d["osti"]["explorer"])
+        elsevier = ConnectionModel.parse_obj(d["elsevier"])
+        osti = OSTIModel(elink=elink, explorer=explorer)
+        materials_store = cls._create_mongostore(config=d, config_collection_name="materials_collection")
+        robocrys_store = cls._create_mongostore(config=d, config_collection_name="robocrys_collection")
+        doi_store = cls._create_mongostore(config=d, config_collection_name="dois_collection")
```

Comment:
The maggma `Store` is already MSONable, so you can just use `MontyDecoder().process_decoded(v)`